doc/BUILD

```diff
@@ -229,6 +229,53 @@ py_test_run_all_subdirectory(
     tags = ["exclusive", "team:core"],
 )
 
+# --------------------------------------------------------------------
+# Test all Workspace template notebooks in doc/source/templates
+# as smoke tests.
+# --------------------------------------------------------------------
+
+filegroup(
+    name = "workspace_templates",
+    srcs = glob(["source/templates/tests/*.ipynb"]),
+    visibility = ["//doc:__subpackages__"]
+)
+
+# Validate that all the paths and yamls within the templates.yaml file are valid.
+
+py_test(
+    name = "templates_directory_validation",
+    size = "small",
+    main = "source/templates/validate.py",
+    srcs = ["source/templates/validate.py"],
+    data = glob(["source/templates/**/*"]),
+    tags = ["exclusive", "team:ml"]
+)
+
+# Templates that require GPU
+
+py_test_run_all_notebooks(
+    size = "large",
+    include = [
+        "source/templates/tests/batch_inference.ipynb",
+        "source/templates/tests/serving_stable_diffusion.ipynb"
+    ],
+    exclude = [],
+    data = ["//doc:workspace_templates"],
+    tags = ["exclusive", "team:ml", "ray_air", "gpu"],
+    env = {"SMOKE_TEST": "1"},
+)
+
+# Templates that require timeseries libs
+
+py_test_run_all_notebooks(
+    size = "large",
+    include = ["source/templates/tests/many_model_training.ipynb"],
+    exclude = [],
+    data = ["//doc:workspace_templates"],
+    tags = ["exclusive", "team:ml", "ray_air", "timeseries_libs"],
```

Comment:
Do we need to include `timeseries_libs` here or should we just install it via runtime environments?

Reply:
Yeah, I think this is no longer needed. I think I needed this before because I was doing a `pip install` and using `statsforecast` stuff outside of remote functions w/ the runtime env. But, installing within the notebook doesn't make the packages immediately available later in the program, so it would fail...